src/vehicle.cpp

```diff
@@ -3193,6 +3205,43 @@ int vehicle::basic_consumption(const itype_id &ftype) const
     return fcon;
 }
 
+int vehicle::supplemental_consumption( const itype_id &ftype ) const
+{
+    int fcon = 0;
+    if( chainsaw_on ) {
+        std::vector<int> all_chainsaws = all_parts_with_feature("CHAINSAW");
+        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
+            if( part_info( all_chainsaws[c] ).fuel_type == ftype ) {
+                fcon += part_power( all_chainsaws[c] );
+                if( one_in( 8 ) ) {
+                    tripoint part_pos = global_pos3() + parts[ all_chainsaws[c] ].precalc[0];
+                    //~ sound of working chainsaw
+                    sounds::sound(part_pos, 20, _( "vrr-vrr-vroom!" ));
```

Comment:
I can't come up with a better onomatopoeia than that. Everything I can think of is too ridiculous looking.